### PR TITLE
Experiment: Unsuspend all lanes on update

### PR DESCRIPTION
**Pull request by acdlite**
Adds a feature flag to tweak the internal heuristic used to "unsuspend" lanes when a new update comes in.

A lane is "suspended" if we couldn't finish rendering it because it was missing data, and we chose not to commit the fallback. (In this context, "suspended" does not include updates that finished with a fallback.)

When we receive new data in the form of an update, we need to retry rendering the suspended lanes, since the new data may have unblocked the previously suspended work. For example, the new update could navigate back to an already loaded route.

It's impractical to retry every combination of suspended lanes, so we need some heuristic that decides which lanes to retry and in which order.

The existing heuristic roughly approximates the old Expiration Times model. It unsuspends all lower priority lanes, but leaves higher priority lanes suspended.

Then when we start rendering, we choose the lanes that have the highest LanePriority and render those -- and then we add to that all the lanes that are higher priority.

If this sounds terribly confusing, it's because it barely makes sense. (It made more sense in the Expiration Times world, I promise, but it was still confusing.) I don't think it's worth me trying to explain the old behavior too much because the point here is that we can replace it with something simpler.

The new heuristic is to unsuspend all suspended lanes whenever there's an update.

This is effectively what we already do except in a few very specific edge cases, ever since we removed the delayed suspense feature from everything that's not a refresh transition.

We can optimize this in the future to only unsuspend lanes that are either 1) in the `lanes` or `subtreeLanes` of the node that was updated, or 2) in the `lanes` of the return path of the node that was updated. This would exclude lanes that are only located in unrelated sibling trees. But, this optimization wouldn't be useful currently because we assign the same transition lane to all transitions. It will become relevant again once we start assigning arbitrary lanes to transitions -- but that in turn requires us to implement entanglement of overlapping transitions, one of our planned projects.

So to sum up: the goal here is to remove the weird edge cases and switch to a simpler model, on top of which we can make more substantial improvements.

I put it behind a flag so I can run an A/B test and confirm it doesn't cause a regression.

### CODE REVIEW

File: packages/react-reconciler/src/__tests__/ReactExpiration-test.js [hunk 1]
```diff
@@ -733,13 +733,19 @@ describe('ReactExpiration', () => {
       // Both normal pri updates should have expired.
       expect(Scheduler).toFlushExpired([
         'Sibling',
-        // Note: we also flushed the high pri update here, because in the
-        // current implementation, once we pick the next lanes to work on, we
-        // entangle it with all pending at equal or higher priority. We could
-        // feasibly change this heuristic so that the high pri update doesn't
-        // render until after the expired updates have finished. But the
-        // important thing in this test is that the normal updates expired.
```
Comment (acdlite) [inline packages/react-reconciler/src/__tests__/ReactExpiration-test.js:741]: I love when a test fails and there's a comment that anticipates the exact reason. Thank you to the very smart, cool, and interesting person who wrote it! (me if that wasn't clear)

File: packages/react-reconciler/src/__tests__/ReactSuspenseList-test.js [hunk 2]
```diff
@@ -2443,41 +2443,30 @@ describe('ReactSuspenseList', () => {
 
     expect(ReactNoop).toMatchRenderedOutput(null);
 
-    ReactNoop.act(() => {
+    await ReactNoop.act(async () => {
```
Comment (acdlite) [inline packages/react-reconciler/src/__tests__/ReactSuspenseList-test.js:2446]: I rewrote this test to decouple it from the weird current behavior of discrete updates (async for a bit, unless another event  comes in). Passes in both implementations.

**Comment by bvaughn** [comment]
> If this sounds terribly confusing, it's because it barely makes sense.

😆

**Review by bvaughn — APPROVED** [review]
